cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -571,7 +575,7 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 		typedErr := a.scaleDownPlanner.UpdateClusterState(podDestinations, scaleDownCandidates, actuationStatus, pdbs, currentTime)
 		// Update clusterStateRegistry and metrics regardless of whether ScaleDown was successful or not.
 		unneededNodes := a.scaleDownPlanner.UnneededNodes()
-		a.clusterStateRegistry.UpdateScaleDownCandidates(unneededNodes, currentTime)
+		a.ScaleDownCandidatesObservers.Update(unneededNodes, currentTime)
 		metrics.UpdateUnneededNodesCount(len(unneededNodes))
```

Comment:
nit: Metric updating is yet another candidate for an observer of unneeded nodes.

Reply:
Yes, but this requires creating an object instead of just calling function, not sure we need it.

Reply:
I think separating metrics reporting from the rest of the logic would be beneficial, we'd get a nice separation of concerns, making the code more testable. That'd require much more than just removing a single metric update, so yeah, I guess it's not very beneficial on its own.